utils/dynamicip/dynamicip.go

```diff
@@ -0,0 +1,166 @@
+package dynamicip
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/logging"
+)
+
+type DynamicResolver interface {
+	Resolve() (string, error)
+	IsResolver() bool
+}
+
+type NoResolver struct {
+}
+
+func (r *NoResolver) IsResolver() bool {
+	return false
+}
+
+func (r *NoResolver) Resolve() (string, error) {
+	return "", errors.New("invalid resolver")
+}
+
+type OpenDNSResolver struct {
+	*net.Resolver
+}
+
+func NewOpenDNSResolver() *OpenDNSResolver {
+	return &OpenDNSResolver{&net.Resolver{
+		PreferGo: true,
+		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+			d := net.Dialer{
+				Timeout: time.Millisecond * time.Duration(10000),
```

Comment:
Can you do `10 * time.Second` instead of `time.Millisecond * time.Duration(10000)`